test/enginetest.c

```diff
@@ -121,8 +121,10 @@ static int test_engines(void)
     display_engine_list();
 
     /*
-     * Depending on whether there's any hardware support compiled in, this
-     * remove may be destined to fail.
+     * This supposedly removes |new_h1|, but that only works if no extra
+     * engines have been loaded, such as the dynamic engine that gets
+     * loaded when dealing with a config file.
+     * See global_init() below for how we deal with the issue.
```

Comment:
I'm not sure I understand this comment or the purpose of the code below. Doesn't `new_h1` get removed by the code on lines 80-82 above?

Reply:
Ah, no, not when rdrand or somesuch happens to be loaded...  It seems like this comment should simply be removed, and only the one in global_init() should remain.  Sounds about right?

Reply:
Ok...but do they only get loaded via config file? In which case do we need this code at all if you have the global_init() code?

Reply:
I'll try to clarify with a reworded comment